src/sentry/ingest/consumer_v2/factory.py

```diff
@@ -0,0 +1,133 @@
+    def __init__(
+        self,
+        function: Callable[[Message[KafkaPayload]], None],
+        multi_process: Optional[MultiProcessConfig] = None,
+    ):
+        self.function = function
+        self.multi_process = multi_process
+
+    def create_with_partitions(
+        self,
+        commit: Commit,
+        partitions: Mapping[Partition, int],
+    ) -> ProcessingStrategy[KafkaPayload]:
+        if (mp := self.multi_process) is not None:
+            return RunTaskWithMultiprocessing(
+                self.function,
+                CommitOffsets(commit),
+                mp.num_processes,
+                mp.max_batch_size,
+                mp.max_batch_time,
+                mp.input_block_size,
+                mp.output_block_size,
+                initializer=initialize_consumer_state,
+            )
+        else:
+            return RunTask(
+                function=self.function,
+                next_step=CommitOffsets(commit),
+            )
+
+
+def get_ingest_consumer(
+    topic: str,
+    group_id: str,
+    auto_offset_reset: str,
+    strict_offset_reset: bool,
+    max_batch_size: int,
+    max_batch_time: int,
+    processes: int,
+    input_block_size: int,
+    output_block_size: int,
+    force_topic: str | None,
+    force_cluster: str | None,
+    **options: Any,
+) -> StreamProcessor[KafkaPayload]:
+    topic = force_topic or topic
+
+    consumer_config = get_config(
+        topic,
+        group_id,
+        auto_offset_reset=auto_offset_reset,
+        strict_offset_reset=strict_offset_reset,
+        force_cluster=force_cluster,
+    )
+    consumer = KafkaConsumer(consumer_config)
+
+    # The `attachments` topic that is used for "complex" events needs ordering
+    # guarantees: Attachments have to be written before the event using them
+    # is being processed. We will use a simple serial `RunTask` for those
+    # for now.
+    # For all other topics, we can use multi processing.
+    allow_multi_processing = topic != settings.KAFKA_INGEST_ATTACHMENTS
+    multi_process = None
+    if processes > 1 and allow_multi_processing:
+        multi_process = MultiProcessConfig(
+            num_processes=processes,
+            max_batch_size=max_batch_size,
+            max_batch_time=max_batch_time,
+            input_block_size=input_block_size,
+            output_block_size=output_block_size,
+        )
+
+    processor_factory = IngestStrategyFactory(process_ingest_message, multi_process=multi_process)
+
+    return StreamProcessor(
+        consumer=consumer,
+        topic=Topic(topic),
+        processor_factory=processor_factory,
+        commit_policy=ONCE_PER_SECOND,
+    )
+
+
+def get_config(
+    topic: str,
+    group_id: str,
+    auto_offset_reset: str,
+    strict_offset_reset: bool,
+    force_cluster: str | None,
+) -> MutableMapping[str, Any]:
+    cluster_name: str = force_cluster or settings.KAFKA_TOPICS[topic]["cluster"]
+    create_topics(cluster_name, [topic])
```

Comment:
I don't think you need this line, all the topics are created in the devserver now